### PR TITLE
Template field indexes not checked

### PR DESCRIPTION
**Pull request by evintila**
Added checks against MAX_TEMPLATE_FIELDS when adding template fields.
Added check against the current number of fields when getting a field's info.
Throw exceptions in case any of the checks fail.

### CODE REVIEW

File: src/modules/packet/Template.h [hunk 1]
```diff
@@ -41,7 +43,7 @@ class Template
 
 		// the packet classes for which each field is valid. This will
 		// be checked against the packet classification before exporting
-		uint32_t fieldValidPacketClasses[MAX_TEMPLATE_FIELDS];
+		uint32_t  fieldValidPacketClasses[MAX_TEMPLATE_FIELDS];
```
Comment (nickbroon) [inline src/modules/packet/Template.h]: unnecessary whitespace change.

File: src/modules/packet/Template.h [hunk 2]
```diff
@@ -60,27 +62,53 @@ class Template
 		// Add a template field that takes data from within the packet
 		void addFieldWithOffset(const InformationElement::IeInfo& ie, uint16_t offset, uint16_t header, uint32_t validPacketClasses)
```
Comment (nickbroon) [inline src/modules/packet/Template.h]: I'd suggest to move the implementation of this method to Template.cpp as it has now become more complex.
I don't think the header file is any longer the place for this.

File: src/modules/packet/Template.h [hunk 3]
```diff
@@ -60,27 +62,53 @@ class Template
...
+			else
+			{
+				// the maximum number of template fields has been exceeded -> abort
+				THROWEXCEPTION("exceeded maximum number of template fields (%d)", MAX_TEMPLATE_FIELDS);
+			}
 		};
 
 		// Add a template "meta-"field which gets its data not from the captured packet itself
 		// "length" parameter must still be correct! (i.e. 4 bytes for FT_NUMPACKETS etc.)
 		void addFieldWithoutOffset(const InformationElement::IeInfo& ie)
```
Comment (nickbroon) [inline src/modules/packet/Template.h]: Ditto

File: src/modules/packet/Template.h (hunk 2 above)
Comment (ogasser) [reply src/modules/packet/Template.h]: I agree.

And while we are at it, there is quite some duplicate code in `addFieldWithOffset` and `addFieldWithoutOffset`. To reduce this duplicate code the latter function could call the former.

File: src/modules/packet/Template.cpp [hunk 4]
```diff
@@ -15,8 +15,6 @@
 #include <fstream>
 #include <iostream>
 
-#include "common/msg.h"
```
Comment (ogasser) [inline src/modules/packet/Template.cpp]: I guess this change is not necessary, and the include can stay in the cpp file.

Comment (evintila) [reply src/modules/packet/Template.cpp]: You are right. I moved it because I was using the exception macro defined in the msg.h in the header, which is no longer the case.

Comment (ogasser) [reply src/modules/packet/Template.cpp]: Please put it in the same position such that there is no diff visible.

File: src/modules/packet/Template.h [hunk 5]
```diff
@@ -46,42 +46,23 @@ class Template
 	public:
 		Template(uint16_t id) : templateID(id), fieldCount(0)
 		{
-		};
```
Comment (ogasser) [inline src/modules/packet/Template.h]: Are these changes necessary?

File: src/modules/packet/Template.h [hunk 6]
```diff
@@ -46,42 +46,23 @@ class Template
...
+		}
 
 		~Template()
 		{
-		};
```
Comment (ogasser) [inline src/modules/packet/Template.h]: Are these changes necessary?

File: src/modules/packet/Template.h [hunk 7]
```diff
@@ -46,42 +46,23 @@ class Template
...
+		}
 
 		int getFieldCount() const
 		{
 			return fieldCount;
-		};
```
Comment (ogasser) [inline src/modules/packet/Template.h]: Are these changes necessary?

**Comment by ogasser** [comment]
LGTM :+1:
